src/main/java/uk/gov/companieshouse/authcodenotification/service/PrivateDataRetrievalService.java

```diff
@@ -32,12 +32,18 @@ public OverseasEntityDataApi getOverseasEntityData(String requestId, String comp
                     .execute()
                     .getData();
 
-            ApiLogger.infoContext(requestId, "Successfully retrieved overseas entity data from database",  logDataMap.getLogMap());
+            ApiLogger.infoContext(requestId, "Successfully retrieved overseas entity data from database", logDataMap.getLogMap());
             return overseasEntityDataApi;
-        } catch (URIValidationException | IOException e) {
-            var message = "Error retrieving overseas entity data from database";
-            ApiLogger.errorContext(requestId, message, e, logDataMap.getLogMap());
-            throw new ServiceException(e.getMessage(), e);
+        } catch (URIValidationException | ApiErrorResponseException e) {
```

Comment:
the instanceOf check feels wrong here, I would split the catch into 2 catch blocks and for the common 'throwing' sections, we could have a private method 'ServiceException buildServiceException(String message)' that could be called in each catch block where needed and the returned ServiceException can be thrown from the catch. (same comment applies to the PublicDataRetrievalService)

Reply:
I would agree with the 2 catch blocks.

Reply:
Separated into two catch blocks.